### PR TITLE
Completed express-crud

### PR DESCRIPTION
**Pull request by shawnkost**
Screenshot 2024-06-03 at 11 54 35 AM
Screenshot 2024-06-03 at 11 55 22 AM
Screenshot 2024-06-03 at 11 56 07 AM
Screenshot 2024-06-03 at 11 57 04 AM
Screenshot 2024-06-03 at 11 57 30 AM

### CODE REVIEW

File: express-crud/server.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+});
+
+function checkGradeId(gradeId: string): void {
+  const id = Number(gradeId);
+  if (!Number.isInteger(id)) {
+    throw new ClientError(400, 'id must be an integer');
+  }
+}
+
+function checkGradeContent(grade: unknown): void {
```
Comment (RobertGardner) [inline express-crud/server.ts:128]: I probably would have declared this as a Grade and done the `+score` above in the call.

It doesn't really matter, but I would find it more clear that I'm checking a Grade.

File: express-crud/server.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    const { gradeId } = req.params;
+    checkGradeId(gradeId);
+    const sql = `
+      select * from "grades"
+      where "gradeId" = $1
+      `;
+    const params = [gradeId];
+    const result = await db.query(sql, params);
+    const student = result.rows[0];
+    if (!student)
```
Comment (RobertGardner) [inline express-crud/server.ts:47]: `{}`

If your if-statement runs over 2 lines, it should definitely have {}.

**Review by RobertGardner — APPROVED** [review]

File: express-crud/server.ts (hunk 1 above)
Comment (shawnkost) [reply express-crud/server.ts:128]: Good point!

File: express-crud/server.ts (hunk 2 above)
Comment (shawnkost) [reply express-crud/server.ts:47]: Yeah it's technically on one line, but since I use a vertical monitor, the formatting gets strange at times.

**Pull request merged**